provider/vdpa_sim_blk_utils.py

```diff
@@ -75,16 +64,70 @@ def remove_vdpa_blk_dev(self, name):
         if cmd_result.exit_status == 0:
             raise VDPABlkSimulatorError("The vdpa block %s still exist" % name)
 
-    def setup(self):
+    def setup(self, opts={}):
         """
         Setup vDPA BLK Simulator environment
+        Example: setup({"vdpa-sim-blk": "shared_backend=1"})
         """
-        LOG.debug("Loading vDPA Kernel modules...")
-        self.load_modules()
+        LOG.debug("Loading vDPA Kernel modules..%s", self._modules)
+        self.load_modules(opts)
 
     def cleanup(self):
         """
         Cleanup vDPA BLK Simulator environment
         """
         self.unload_modules()
         LOG.info("vDPA Simulator environment recover successfully.")
+
+
+class VhostVdpaSimulatorTest(VDPABlkSimulatorTest):
+    def __init__(self):
+        super(VhostVdpaSimulatorTest, self).__init__()
+        self._modules = ['vhost-vdpa', 'vdpa-sim-blk']
+
+    def add_vdpa_blk_dev(self, name):
+        """
+        Add vDPA blk device
+        :param name: device name
+        :return : host device name ,eg. /dev/vhost-vdpa-X
+        """
+
+        cmd = "vdpa dev add mgmtdev vdpasim_blk name %s" % name
+        process.run(cmd, shell=True)
+        cmd = "vdpa dev list -jp %s" % name
+        process.run(cmd, shell=True)
+
+        time.sleep(2)
+        try:
+            dev = get_image_filename(name).replace("vdpa://", "")
+        except Exception as e:
+            raise VDPABlkSimulatorError("vdpa dev add %s failed:%s" % (name, e))
```

Comment:
Suggest to use str(e) instead of e. Thanks.

Reply:
updated